Sources/StreamChat/APIClient/CDNClient/CDNClient.swift

```diff
@@ -81,20 +76,28 @@ class StreamCDNClient: CDNClient {
                 log.warning("Callback called while self is nil", subsystems: .httpRequests)
                 return
             }
-            
+
             let task = self.session.dataTask(with: urlRequest) { [decoder = self.decoder] (data, response, error) in
                 do {
                     let decodedResponse: FileUploadPayload = try decoder.decodeRequestResponse(
                         data: data,
                         response: response,
                         error: error
                     )
-                    completion(.success(decodedResponse.file))
+
+                    let uploadedFile = UploadedFile(
+                        remoteURL: decodedResponse.fileURL,
+                        remotePreviewURL: decodedResponse.previewURL
+                    )
+
+                    // TODO: For v5, CDNClient should return the `UploadedFile` directly.
+                    // Especially if we want to support video thumbnails.
```

Comment:
Wasn't the video thumbnail support part of the CDN implementation?

Reply:
The only thing missing is the thumbnail for videos, which at the moment on iOS it is not a priority since we already have it locally. So a separate PR should handle that in the future